mlflow/models/python_api.py

```diff
@@ -178,13 +195,41 @@ def _serialize_input_data(input_data, content_type):
             # first row to be the header
             return pd.DataFrame(input_data).to_csv(index=False)
         else:
-            return json.dumps(input_data)
+            return _serialize_to_json(input_data)
     except Exception as e:
         raise MlflowException.invalid_parameter_value(
-            message="Input data could not be serialized to {content_type}."
+            message=f"Input data could not be serialized to {content_type}."
         ) from e
 
 
+def _serialize_to_json(input_data):
+    # imported inside function to avoid circular import
+    from mlflow.pyfunc.scoring_server import SUPPORTED_FORMATS, SUPPORTED_LLM_FORMAT
+
+    if isinstance(input_data, dict) and any(
+        key in input_data for key in SUPPORTED_FORMATS | {SUPPORTED_LLM_FORMAT}
+    ):
+        return json.dumps(input_data)
+    else:
+        original_input_data = input_data
+
+        if isinstance(input_data, (datetime, bool, bytes, float, int, str)):
+            input_data = [input_data]
+
+        input_data = dump_input_data(input_data)
+
+        _logger.warning(
```

Comment:
should we have this at `info` level so that it's easier to disable without blanket disabling other warnings?

Reply:
makes sense, will change it to info